### DISCUSSION TITLE
First release is available

### DISCUSSION THREAD

**Discussion by Beakerboy** [Beakerboy/VBA-Linter#52] [Announcements]
@DecimalTurn

This project has an official release. You can use it in the command line as:

```
pip install vba_linter
python -m vba_linter ./src
```

or you can use the action as seen here:
https://github.com/Beakerboy/VBA-SQL-Library/blob/Dev-action/.github/workflows/lint_vba.yml

Let me know if you have any feedback. I need to make the maximum line length configurable for sure.

**Comment by DecimalTurn** [comment]
Thanks for the heads up, that looks promising!

Regarding the possibility to configure certain settings, have you considered using an [EditorConfig](https://editorconfig.org/) file? There is a convention for [max_line_length](https://github.com/editorconfig/editorconfig/wiki/editorconfig-properties#max_line_length) in there that could be used to customize the maximum line length check.

Here is an example of what the EditorConfig file could look like for a VBA project: 
https://github.com/DecimalTurn/Excel-Pomodoro-Timer/blob/main/.editorconfig

The nice thing with EditorConfig is that it's widely used and there is already a parser written for python (https://github.com/editorconfig/editorconfig-core-py). Furthermore, it supports [line endings](https://github.com/editorconfig/editorconfig/wiki/editorconfig-properties#end_of_line), [trailing whitespace](https://github.com/editorconfig/editorconfig/wiki/editorconfig-properties#trim_trailing_whitespace) and more.

**Reply by Beakerboy** [reply]
I have not seen that tool before, thanks. As a first pass I was going to see how flake8 for python does things, however, there are fewer competing formatting standards in python than other languages I’ve used, so flake8 and pycodestyle might not be very configurable.

**Reply by DecimalTurn** [reply]
The `.editorconfig` file is just a configuration file, so you could simply keep your current approach and create a function that reads the desired settings for the repo in the `.editorconfig` file.

For instance, you could replace the following line in https://github.com/Beakerboy/VBA-Linter/blob/main/src/vba_linter/rules/w501.py#L12C11-L12C11
```
self._max_len = 79
```

with something like 
```
self._max_len = get_editorconfig_value("max_line_length")
```

where get_editorconfig_value would have to be implemented using [editorconfig-core-py](https://github.com/editorconfig/editorconfig-core-py)

**Reply by Beakerboy** [reply]
Yes, I got that. In theory the there could be an abstracted config object interface and multiple implementations that take in the configuration options any number of ways. Like I said I was going to first see how flake8 handle the customizations.

**Reply by DecimalTurn** [reply]
Ok, I don't know how flake8 customizations works, would that be specified as parameters inside the GitHub action itself?

**Reply by Beakerboy** [reply]
I don’t know…I have looked at it yet. I know at least one can be passed in as a command line option, and I’m assuming some can be passed in through the pyproject.toml file.

**Reply by DecimalTurn** [reply]
But the `pyproject.toml` would have to reside inside the repo that is being linted for that to work, no? Wouldn't that be weird to have a `pyproject.toml` file inside a VBA project?

**Reply by Beakerboy** [reply]
Yes…obviously I wouldn't do that, but I could use the formatting and naming of the options as a guide.

**Reply by Beakerboy** [reply]
If you have a particular path you would like to see on this, feel free to put something together in a pull request. As for me, I’m going to investigate how a few other projects do things and make a decision.

**Reply by DecimalTurn** [reply]
I've done some work on this [here](https://github.com/DecimalTurn/VBA-Linter/pull/1), but I'm not familiar with all this, so any advice would be really appreciated.

**Reply by Beakerboy** [reply]
Would you mind re-rolling this against the dev branch? This is where I’m staging the changes for the next release. Thanks.

**Reply by DecimalTurn** [reply]
Sure, here it is https://github.com/Beakerboy/VBA-Linter/pull/67
